### ISSUE TITLE
Fix Tpetra configuration error message.

### ISSUE THREAD

**Issue by bangerth** [dealii/dealii#16928]
In `cmake/configure/configure_20_trilinos.cmake` we have
```
        if(NOT DEAL_II_HAVE_TPETRA_INT_LONG_LONG AND DEAL_II_WITH_64BIT_INDICES)
          message(
            STATUS
            "  Tpetra wasn't configured with support for 64-bit global indices\n"
            "  but deal.II is configured to use 64-bit global indices.\n"
            "  Either reconfigure deal.II with -DDEAL_II_WITH_64BIT_INDICES=OFF\n"
            "  or rebuild Trilinos with -DTPETRA_INST_INT_LONG_LONG=ON"
          )
          if(DEAL_II_HAVE_TPETRA_INT_LONG)
            message(
              STATUS
              "  and -DTPETRA_INST_INT_LONG=OFF."
            )
          elseif(DEAL_II_HAVE_TPETRA_INT_INT)
            message(
              STATUS
              "  and -DTPETRA_INST_INT_INT=OFF."
            )
          endif()
        elseif(NOT DEAL_II_HAVE_TPETRA_INT_INT AND NOT DEAL_II_WITH_64BIT_INDICES)
          message(
            STATUS
            "  Tpetra wasn't configured with support for 32-bit global indices\n"
            "  but deal.II is configured to use 32-bit global indices.\n"
            "  Either reconfigure deal.II with -DDEAL_II_WITH_64BIT_INDICES=ON\n"
            "  or rebuild Trilinos with -DTPETRA_INST_INT_INT=ON"
          )
          if(DEAL_II_HAVE_TPETRA_INT_LONG)
            message(
              STATUS
              "  and -DTPETRA_INST_INT_LONG=OFF."
            )
          elseif(DEAL_II_HAVE_TPETRA_INT_LONG_LONG)
            message(
              STATUS
              "  and -DTPETRA_INST_INT_LONG_LONG=OFF."
            )
```
These error messages suggest doing `-DTPETRA_INST_INT_INT=ON` etc, but Trilinos does not actually understand this flag. (This is true at least for Trilinos 15.) I don't know what the currently correct invocation of Trilinos cmake is, but this is not it :-)

**Comment by bangerth** [comment]
To wit:
```
> grep -r TPETRA_INST_INT_INT CMake*                                          
CMakeCache.txt:HAVE_MUELU_TPETRA_INST_INT_INT:INTERNAL=OFF
CMakeCache.txt:HAVE_TPETRA_INST_INT_INT:INTERNAL=OFF
CMakeCache.txt:HAVE_TPETRA_INST_INT_INT_DEFAULT:INTERNAL=OFF
CMakeCache.txt:Tpetra_PKG_VARS_TO_EXPORT:INTERNAL=Tpetra_ENABLE_DEBUG;HAVE_TPETRA_DEBUG;Tpetra_ENABLE_DEPRECATED_CODE;TPETRA_ENABLE_DEPRECATED_CODE;Tpetra_IGNORE_KOKKOS_COMPATIBILITY;Tpetra_ENABLE_BLOCKCRS_LITTLEBLOCK_LAYOUTLEFT;TPETRA_ENABLE_BLOCKCRS_LITTLEBLOCK_LAYOUTLEFT;Tpetra_ENABLE_CUDA;TPETRA_ENABLE_CUDA;Tpetra_INST_SERIAL;HAVE_TPETRA_INST_SERIAL;Tpetra_INST_OPENMP;HAVE_TPETRA_INST_OPENMP;Tpetra_INST_PTHREAD;HAVE_TPETRA_INST_PTHREAD;Tpetra_INST_CUDA;HAVE_TPETRA_INST_CUDA;Tpetra_INST_HIP;HAVE_TPETRA_INST_HIP;Tpetra_INST_SYCL;HAVE_TPETRA_INST_SYCL;Tpetra_ASSUME_CUDA_AWARE_MPI;TPETRA_ASSUME_CUDA_AWARE_MPI;Tpetra_ASSUME_GPU_AWARE_MPI;TPETRA_ASSUME_GPU_AWARE_MPI;Tpetra_INST_INT_INT;HAVE_TPETRA_INST_INT_INT;Tpetra_INST_INT_UNSIGNED;HAVE_TPETRA_INST_INT_UNSIGNED;Tpetra_INST_INT_LONG_LONG;HAVE_TPETRA_INST_INT_LONG_LONG;Tpetra_INST_INT_LONG;HAVE_TPETRA_INST_INT_LONG;Tpetra_INST_INT_UNSIGNED_LONG;HAVE_TPETRA_INST_INT_UNSIGNED_LONG;Tpetra_INST_DOUBLE;HAVE_TPETRA_INST_DOUBLE;Tpetra_INST_FLOAT;HAVE_TPETRA_INST_FLOAT;Tpetra_INST_LONG_DOUBLE;HAVE_TPETRA_INST_LONG_DOUBLE;Tpetra_INST_COMPLEX_DOUBLE;HAVE_TPETRA_INST_COMPLEX_DOUBLE;Tpetra_INST_COMPLEX_FLOAT;HAVE_TPETRA_INST_COMPLEX_FLOAT;Tpetra_INST_FLOAT128;HAVE_TPETRA_INST_FLOAT128;Tpetra_INST_QD_REAL;HAVE_TPETRA_INST_QD_REAL;Tpetra_INST_DD_REAL;HAVE_TPETRA_INST_DD_REAL;Tpetra_ENABLE_Experimental;HAVE_TPETRA_EXPERIMENTAL
```
The only places where `TPETRA_INST_INT_INT` appears is with the `HAVE_` prefix. I figure that these are internally computed variables, rather than ones we should set on the command line.

**Comment by masterleinad** [comment]
It's `Tpetra_INST_INT_INT`.

**Comment by bangerth** [comment]
Oh, you beat me to it. I realized that too now. We just have to fix the spelling in the cmake file. Let's postpone until after #16929 to avoid the merge conflict.
